### PR TITLE
Add retries to Resource PDF Generation

### PR DESCRIPTION
**Pull request by Hamms**
For background, we've had an intermittent issue with the Curriculum PDF generation process in which when it comes time to collate all our individual PDFs together in the rollup, GhostScript complains that it can't find all the files that we're telling it to roll up, and the PDF generation fails.

I've been able to track down the failure to specifically the title pages that we generate in the Resource rollup to differentiate the lessons within a script. Unfortunately, I have been unable to get any kind of reliable repro, much less to figure out *why* this is happening in the first place.

Because I know where the problem is but not why it is, and because it happens so rarely as to make further investigation difficult, I'm resorting to a retry. It's not a good solution, but it's the best I got at this point.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
seems quite reasonable!

File: dashboard/lib/services/curriculum_pdfs/resources.rb [hunk 1]
```diff
@@ -113,21 +113,24 @@ def generate_lesson_resources_title_page(lesson, directory="/tmp/")
 
           # we've been having some issues with these title page PDFs not
           # existing on the filesystem when it comes time to make the rollup.
+          #
           # It's not yet clear whether that's because this step is failing to
-          # generate or because the file is getting vanished after generation.
-          # Adding some logging here to help diagnose.
-          unless $?.success?
-            ChatClient.log(
-              "PDF generation exited with status code #{$?.exitstatus.inspect}",
```
Comment (bethanyaconnor) [inline dashboard/lib/services/curriculum_pdfs/resources.rb:121]: Did you get any helpful error information here? If so, can you summarize what it was before removing it?

**Review by bethanyaconnor — APPROVED** [review]

File: dashboard/lib/services/curriculum_pdfs/resources.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/lib/services/curriculum_pdfs/resources.rb:121]: I did and I didn't. This clause never triggered at any point, so in once sense I got no information, and in another sense I learned that even when we end up failing to generate a file (as indicated by the other clause triggering), the process still _reports_ a success. So I'm removing the clause, but it is worth keeping in mind that whatever's going on here, it's subtle.

Comment (bethanyaconnor) [reply dashboard/lib/services/curriculum_pdfs/resources.rb:121]: That's what I figured, but wanted to double check. Thanks!
